### PR TITLE
Add prerelease workflow; update workflows; update pyproject

### PR DESCRIPTION
**Pull request by bonicim**
* Add pre-release workflow
* Update CodeQL actions per https://github.blog/changelog/2022-04-27-code-scanning-deprecation-of-codeql-action-v1/
* Update pyproject

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -1,13 +1,16 @@
 [build-system]
-requires = ["poetry-core>=1.1.0a6"]
+requires = ["poetry-core>=1.2.0b2"]
```
Comment (craig8) [inline pyproject.toml:2]: change to using 1.2 rather than b2 as 1.2 is out now.

File: pyproject.toml [hunk 2]
```diff
@@ -1,13 +1,16 @@
...
 build-backend = "poetry.core.masonry.api"
 
 [tool.mypy]
+python_version = 3.9
```
Comment (craig8) [inline pyproject.toml:6]: Should this be the minimum we support?

**Review by craig8 — CHANGES_REQUESTED** [review]
Looks good, I have just a couple comments that should be looked at but very good

File: pyproject.toml (hunk 2 above)
Comment (bonicim) [reply pyproject.toml:6]: Will change to 3.8 to align with the python minimum dependency in [tool.poetry.dependencies].

File: pyproject.toml (hunk 1 above)
Comment (bonicim) [reply pyproject.toml:2]: Looks like poetry just updated to [1.2.2](https://python-poetry.org/blog/category/releases/). Will update and test locally before pushing a new commit.

**Pull request merged**